src/tools.rs

```diff
@@ -185,3 +186,66 @@ pub fn parse_maps(pid: u32) -> Result<Vec<LinuxMapsEntry>, Error> {
 
     Ok(entries)
 }
+
+struct SyncQueueSharedState<T> {
```

Comment:
why implementing such generic things, I'm sure Rust ecosystem already solved this problem. Have you looked at https://crates.io/crates/crossbeam-channel for example? Any reasons not to use some cargo package instead?

Reply:
I have replaced it with the crossbeam-channel